examples/api/lib/material/toggle_buttons/toggle_buttons.1.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+/// Flutter code sample for migrating from [ToggleButtons] to [SegmentedButton].
+
+void main() {
+  runApp(const ToggleButtonsApp());
+}
+
+class ToggleButtonsApp extends StatelessWidget {
+  const ToggleButtonsApp({super.key});
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      theme: ThemeData(useMaterial3: true),
+      home: const Scaffold(
+        body: ToggleButtonsExample(),
+      ),
+    );
+  }
+}
+
+enum Sizes { extraSmall, small, medium, large, extraLarge }
```

Comment:
Seeing `Sizes` in the code below can be slightly confusing because `Size`. Calling this `ShirtSize` would make that slightly less confusing.

Also: this is a nice opportunity to use the new Record facility https://dart.dev/language/records
```dart
enum ShirtSize { extraSmall, small, medium, large, extraLarge }
const List<(ShirtSize, String)> shirtSizeOptions = [
  (extraSmall, 'XS'),
   // etc
];
```

Then you could create the ToggleButtons children and the SegmentedButtons segments with`shirtSizeOptions.map` expressions.

Reply:
Excellent idea!